File: Dddml.Wms.JavaCommon/src/org/dddml/wms/domain/UserLoginMvoStateEventIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class UserLoginMvoStateEventIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "userLoginIdUserId",
            "userLoginIdLoginKeyLoginProvider",
            "userLoginIdLoginKeyProviderKey",
            "userVersion",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
            "String",
            "Long",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private UserLoginMvoStateEventId value;

    public UserLoginMvoStateEventIdFlattenedDto()
    {
        this(new UserLoginMvoStateEventId());
    }

    public UserLoginMvoStateEventIdFlattenedDto(UserLoginMvoStateEventId value)
    {
        this.value = value;
    }

    public UserLoginMvoStateEventId toUserLoginMvoStateEventId()
    {
        return this.value;
    }


    public String getUserLoginIdUserId()
    {
        return this.value.getUserLoginId().getUserId();
    }

    public void setUserLoginIdUserId(String userLoginIdUserId)
    {
        this.value.getUserLoginId().setUserId(userLoginIdUserId);
    }

    public String getUserLoginIdLoginKeyLoginProvider()
    {
        return this.value.getUserLoginId().getLoginKey().getLoginProvider();
    }

    public void setUserLoginIdLoginKeyLoginProvider(String userLoginIdLoginKeyLoginProvider)
    {
        this.value.getUserLoginId().getLoginKey().setLoginProvider(userLoginIdLoginKeyLoginProvider);
    }

    public String getUserLoginIdLoginKeyProviderKey()
    {
        return this.value.getUserLoginId().getLoginKey().getProviderKey();
    }

    public void setUserLoginIdLoginKeyProviderKey(String userLoginIdLoginKeyProviderKey)
    {
        this.value.getUserLoginId().getLoginKey().setProviderKey(userLoginIdLoginKeyProviderKey);
    }

    public Long getUserVersion()
    {
        return this.value.getUserVersion();
    }

    public void setUserVersion(Long userVersion)
    {
        this.value.setUserVersion(userVersion);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        UserLoginMvoStateEventIdFlattenedDto other = (UserLoginMvoStateEventIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/UserLoginMvoStateEventIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;
use Dddml\Serializer\Type\Long;
use Dddml\FlattenedDtoInterface;

class UserLoginMvoStateEventIdFlattenedDto implements FlattenedDtoInterface
{

    const PROPERTIES = [
            'userLoginIdUserId' => 'string',
            'userLoginIdLoginKeyLoginProvider' => 'string',
            'userLoginIdLoginKeyProviderKey' => 'string',
            'userVersion' => 'Long',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var UserLoginMvoStateEventId
     */
    private $value;

    /**
     * @param UserLoginMvoStateEventId $value
     */
    public function __construct(UserLoginMvoStateEventId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new UserLoginMvoStateEventId();
        }
    }

    /**
     * @return UserLoginMvoStateEventId
     */
    public function toUserLoginMvoStateEventId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getUserLoginIdUserId()
    {
        return $this->value->getUserLoginId()->getUserId();
    }

    /**
     * @param string $userLoginIdUserId
     */
    public function setUserLoginIdUserId($userLoginIdUserId)
    {
        $this->value->getUserLoginId()->setUserId($userLoginIdUserId);
    }

    /**
     * @return string
     */
    public function getUserLoginIdLoginKeyLoginProvider()
    {
        return $this->value->getUserLoginId()->getLoginKey()->getLoginProvider();
    }

    /**
     * @param string $userLoginIdLoginKeyLoginProvider
     */
    public function setUserLoginIdLoginKeyLoginProvider($userLoginIdLoginKeyLoginProvider)
    {
        $this->value->getUserLoginId()->getLoginKey()->setLoginProvider($userLoginIdLoginKeyLoginProvider);
    }

    /**
     * @return string
     */
    public function getUserLoginIdLoginKeyProviderKey()
    {
        return $this->value->getUserLoginId()->getLoginKey()->getProviderKey();
    }

    /**
     * @param string $userLoginIdLoginKeyProviderKey
     */
    public function setUserLoginIdLoginKeyProviderKey($userLoginIdLoginKeyProviderKey)
    {
        $this->value->getUserLoginId()->getLoginKey()->setProviderKey($userLoginIdLoginKeyProviderKey);
    }

    /**
     * @return Long
     */
    public function getUserVersion()
    {
        return $this->value->getUserVersion();
    }

    /**
     * @param Long $userVersion
     */
    public function setUserVersion($userVersion)
    {
        $this->value->setUserVersion($userVersion);
    }

    /**
     * @return string
     */
    public function toString()
    {
        $pValues = [
            $this->getUserLoginIdUserId(),
            $this->getUserLoginIdLoginKeyLoginProvider(),
            $this->getUserLoginIdLoginKeyProviderKey(),
            $this->getUserVersion(),
        ];
        return implode(',', $pValues);
    }

    /**
     * @param string $data
     *
     * @return UserLoginMvoStateEventIdFlattenedDto
     */
    public function fromString($data)
    {
        $pValues = explode(',', $data);
        $this->setUserLoginIdUserId($pValues[0]);
        $this->setUserLoginIdLoginKeyLoginProvider($pValues[1]);
        $this->setUserLoginIdLoginKeyProviderKey($pValues[2]);
        $this->setUserVersion($pValues[3]);
        return $this;
    }

}

